tokio/src/runtime/basic_scheduler.rs

```diff
@@ -57,6 +57,10 @@ struct Core {
 
     /// Metrics batch
     metrics: MetricsBatch,
+
+    /// True if a task panicked without being handled and the runtime is
+    /// configured to shutdown on unhandled panic.
+    unhandled_panic: bool,
```

Comment:
previously i remember thinking that it would be nice to be able to actually store the unhandled panic and propagate it, rather than creating a new panic that loses the original panic's message, but you pointed out that that [doesn't work, because the panic is also shared with the `JoinHandle`](https://github.com/tokio-rs/tokio/pull/4518#issuecomment-1050117245) and they could race for it.

it occurs to me, though, that now that we have task IDs, perhaps we should store the ID of the task that panicked, rather than just a bool indicating that a task panicked. that way, we could at least identify _which_ task panicked when the runtime panics, which could help users debug the panic when combined with other logs/diagnostics. WDYT?